src/Generator/CreditMemoPdfFileGenerator.php

```diff
@@ -28,30 +28,36 @@ final class CreditMemoPdfFileGenerator implements CreditMemoPdfFileGeneratorInte
 
     private RepositoryInterface $creditMemoRepository;
 
-    private Environment $twig;
+    private ?Environment $twig;
 
-    private GeneratorInterface $pdfGenerator;
+    private ?GeneratorInterface $pdfGenerator;
 
     private FileLocatorInterface $fileLocator;
 
     private string $template;
 
     private string $creditMemoLogoPath;
 
+    private ?TwigToPdfGeneratorInterface $twigToPdfGenerator;
+
     public function __construct(
         RepositoryInterface $creditMemoRepository,
-        Environment $twig,
-        GeneratorInterface $pdfGenerator,
+        ?Environment $twig,
+        ?GeneratorInterface $pdfGenerator,
         FileLocatorInterface $fileLocator,
         string $template,
-        string $creditMemoLogoPath
+        string $creditMemoLogoPath,
+        ?TwigToPdfGeneratorInterface $twigToPdfGenerator = null
     ) {
         $this->creditMemoRepository = $creditMemoRepository;
         $this->twig = $twig;
         $this->pdfGenerator = $pdfGenerator;
         $this->fileLocator = $fileLocator;
         $this->template = $template;
         $this->creditMemoLogoPath = $creditMemoLogoPath;
+        $this->twigToPdfGenerator = $twigToPdfGenerator;
```

Comment:
I would be for adding here a new argument, not a `TwigToPdfGeneratorInterface`, but a new service that provides options for generating PDF file. You can then use the current logic and provide options only if the service is available. Because as I see, the logic extracted to `TwigToPdfGeneratorInterface` is mainly preparing options. Then you will need to deprecate only not using the new service.

Reply:
Ok, I've replaced `TwigToPdfGeneratorInterface` with `PdfOptionsGeneratorInterface` and will add `TwigToPdfGeneratorInterface` in the separate PR.